### PR TITLE
added supoort for metrics MDS

### PR DESCRIPTION
**Pull request by sumukhswamy**
### Description
With multiple data source enabled, users of OpenSearch Dashboards with Metrics analytics are able to fetch metrics from local and remote clusters
MDS support for metrics analytics
https://github.com/opensearch-project/dashboards-observability/issues/1754

### Issues Resolved
[List any issues this PR will resolve]

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by YANG-DB — APPROVED** [review]

**Comment by sumukhswamy** [comment]
https://github.com/opensearch-project/dashboards-observability/assets/107440906/b025fe73-8111-4c51-8f7b-ef825cfcaa53

**Comment by sumukhswamy** [comment]
Screenshot 2024-06-07 at 10 39 48 AM
Screenshot 2024-06-07 at 10 40 35 AM

**Review by BionIT — APPROVED** [review]

File: server/routes/metrics/metrics_rounter.ts [hunk 1]
```diff
@@ -113,14 +121,20 @@ export function registerMetricsRoute(router: IRouter) {
       request,
       response
     ): Promise<IOpenSearchDashboardsResponse<any | ResponseError>> => {
-      const opensearchNotebooksClient: ILegacyScopedClusterClient = context.observability_plugin.observabilityClient.asScoped(
-        request
-      );
-
+      const dataSourceMDSId = request.params.dataSourceMDSId;
+      let opensearchNotebooksClient;
```
Comment (ps48) [inline server/routes/metrics/metrics_rounter.ts:125]: Nit. Not sure if this should be named notebooks client

File: public/services/requests/ppl.ts [hunk 2]
```diff
@@ -19,11 +19,15 @@ export default class PPLService {
       query: string;
       format: string;
     },
+    dataSourceMDSId?: string,
```
Comment (ps48) [inline public/services/requests/ppl.ts:22]: With this change do we need to add MDS param here: https://github.com/opensearch-project/dashboards-observability/blob/19d68b2d443458c3fdb1bd0fe035bf823111015b/public/services/timestamp/timestamp.ts#L81C9-L81C28

**Review by ps48 — APPROVED** [review]

File: public/services/requests/ppl.ts (hunk 2 above)
Comment (ps48) [reply public/services/requests/ppl.ts:22]: Discussed with @sumukhswamy offline this isn't being used by metrics analytics.

**Pull request merged**